File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> hm = new HashMap<Integer,Integer>();
        int ans[] = new int[2];
        for(int i = 0; i < nums.length; i++)
        {
            if(hm.containsKey(target - nums[i]))
            {
                ans[0] = i;
                ans[1] = hm.get(target - nums[i]);
            }
            hm.put(nums[i], i);
        }
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
         
        map<int,int> hm;
        vector<int> arr(2);
        for(int i = 0; i < nums.size(); i++)
        {
            if(hm.contains(target - nums[i]))
            {
                arr[0] = i;
                arr[1] = hm[target - nums[i]];
                return arr;
            }
            else
            {
                hm[nums[i]] = i;
            }
        }
        return arr;
    }
};